maintenance/utils/Status.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+class MaintenanceStatus {
+	private $cacheKey = null;
+
+	const STARTED = 'STARTED';
+	const ABORTED = 'ABORTED';
+	const FINISHED = 'FINISHED';
+	const TTL = 7200; // 2h in seconds
+
+	public function __construct( $taskName ) {
+		$this->cacheKey = wfMemcKey( $taskName . '_status' );
+	}
+
+	public function markAsStarted() {
+		$this->setCachedStatus( MaintenanceStatus::STARTED );
```

Comment:
Consider using `self::STARTED` here instead of explicit class name. It makes refactors easier and `grep` output more meaningful while searching for class name. Also, it is a nice analogy to human comprehension of possession, as we refer to things we own as "ours", and we only use explicit names while referring to someone else's property. So `self` indicates that the constant belongs to the current class more clearly that the class' name. :)